subprojects/logging/src/main/java/org/gradle/internal/logging/sink/PlainConsoleDispatchingListener.java

```diff
@@ -42,9 +44,13 @@ public void onOutput(OutputEvent event) {
             // TODO - should attempt to flush the output stream prior to writing to the error stream
             stderrChain.onOutput(event);
         } else {
-            // Write anything that isn't the build failure report to the output stream
+            // Write anything that isn't the build failure report to the output stream if there is a console attached.
             // This is to avoid interleaving of error and non-error output generated at approximately the same time, as the process stdout and stderr may be forwarded on different pipes and so ordering is lost
-            stdoutChain.onOutput(event);
+            if (redirectStderr) {
```

Comment:
should also remove the stuff above that checks the category and for to stderr, on line 40, as the new logic replaces this old (and hacky) logic.

Reply:
👍  Done.